### PR TITLE
Fix Issue #355 #217

### PR DESCRIPTION
**Pull request by VikingsFan**
Use name instead of propertyInfo to lookup

### CODE REVIEW

File: OData/test/UnitTest/System.Web.OData.Test/OData/Builder/Conventions/ODataConventionModelBuilderTests.cs [hunk 1]
```diff
@@ -2671,6 +2671,58 @@ public void GetEdmModel_ThrowsExpcetion_ForRecursiveLoopOfComplexType()
                 "The complex type 'System.Web.OData.Builder.Conventions.RecursiveEmployee' has a reference to itself " +
                 "through the property 'Manager'. A recursive loop of complex types is not allowed.");
         }
+
```
Comment (xuzhg) [inline OData/test/UnitTest/System.Web.OData.Test/OData/Builder/Conventions/ODataConventionModelBuilderTests.cs]: remove

**Comment by xuzhg** [comment]
So far, it looks good to me. However, it will be better to cover more scenarios with test cases.

**Comment by xuzhg** [comment]
Merged. Thanks.

File: OData/src/System.Web.OData/OData/Builder/StructuralTypeConfiguration.cs [hunk 2]
```diff
@@ -268,21 +268,13 @@ public virtual PrimitivePropertyConfiguration AddProperty(PropertyInfo propertyI
             ValidatePropertyNotAlreadyDefinedInDerivedTypes(propertyInfo);
 
             // Remove from the ignored properties
-            if (RemovedProperties.Contains(propertyInfo))
+            if (RemovedProperties.Any(prop => prop.Name.Equals(propertyInfo.Name)))
```
Comment (congysu) [inline OData/src/System.Web.OData/OData/Builder/StructuralTypeConfiguration.cs]: Does this work, if with property name aliasing, such as `[DataMember(Name="abc")]`?

**Pull request closed without merging**